pramen/api/src/main/scala/za/co/absa/pramen/api/DataFormat.scala

```diff
@@ -19,29 +19,62 @@ package za.co.absa.pramen.api
 /** Storage formats supported by the metastore. */
 sealed trait DataFormat {
   def name: String
+
+  def isTransient: Boolean
+
+  def isLazy: Boolean
```

Comment:
Lazy is same as OnDemand. Since transient tables existed already, in order to maintain backwards compatibility, the old behavior is happening when 'transient_eager' is used. In this case `isTransient = true` and `isLazy = false`.

The new on-demand behaviour is when `isTransient = true` and `isLazy = true`.

I'm now thinking renaming `isLazy` to `isOnDemand` for less confusion.